workers/unity/Assets/Tests/EditmodeTests/Correctness/Subscriptions/CommandTests.cs

```diff
@@ -40,6 +42,56 @@ public void SubscriptionSystem_invokes_callback_on_receiving_response()
                 });
         }
 
+        [Test]
+        public void SubscriptionSystem_does_not_invoke_callback_when_sender_is_invalid()
+        {
+            World.Step(world =>
+                {
+                    world.Connection.CreateEntity(EntityId, GetTemplate());
+                })
+                .Step(world =>
+                {
+                    return world.CreateGameObject<CommandStub>(EntityId).Item2.Sender;
+                })
+                .Step((world, sender) =>
+                {
+                    sender.SendTestCommand(GetRequest(), response => throw new AssertionException("Don't call"));
+                })
+                .Step((world, sender) =>
+                {
+                    world.Connection.GenerateResponses<TestCommands.Test.Request, TestCommands.Test.ReceivedResponse>(
+                        ResponseGenerator);
+                    sender.IsValid = false;
```

Comment:
Hmm this shouldn't be allowed from the user API, cc @zeroZshadow @paulbalaji thoughts?

Reply:
I think I'd agree. Should there be a way to directly invalidate command senders, or should this test be indirect by having another reader/writer on the script that we invalidate

Reply:
Actually this may not be needed right? Since the `IsValid` bool is only when the user captures the command sender.

Reply:
My point was more that a user shouldn't be able to set `IsValid` since we should be controlling the lifecycle of those objects.

Reply:
Probably cause its internal and internals are exposed to the test assemblies

Reply:
Its public set! https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/io.improbable.gdk.core/Subscriptions/Reader.cs#L14

Reply:
Raised UTY-2712 to fix this